File: PARK_JAVA/PARK_P3_EXAMEN.java
import java.util.*;

public class PARK_P3_EXAMEN{
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        int numero = 0;
        System.out.println("*************************************");
        System.out.println("       DE ENTEROS A ROMANOS");
        System.out.println("*************************************");
        System.out.println("INGRESA UN NUMERO QUE NO EXCEDA 1001: ");
        numero = scanner.nextInt();

        if(numero > 1001){
            System.out.println("ecxedes la limitante intenta otro numero: ");
            numero = scanner.nextInt();
        }

        String respuesta = enteroRomano(numero);        

        System.out.println("Tu numero es: " + numero + " a Romano es:" +respuesta);

        scanner.close();
    }

    public static String enteroRomano(int numero){
        int resta = numero;
        String resultado = "";

        SortedMap<Integer, String> roman = new TreeMap<>(Collections.reverseOrder());
        roman.put(1000, "M");
        roman.put(900, "CM");
        roman.put(500, "D");
        roman.put(400, "CD");
        roman.put(100, "C");
        roman.put(90, "XC");
        roman.put(50, "L");
        roman.put(40, "XL");
        roman.put(10, "X");
        roman.put(9, "IX");
        roman.put(5, "V");
        roman.put(4, "IV");
        roman.put(1, "I");
       
        Set<Map.Entry<Integer, String>> s = roman.entrySet();
        
        Iterator<Map.Entry<Integer, String>> i = s.iterator();
        
        while(i.hasNext()){
           
            Map.Entry<Integer, String> entrada = i.next();
           
            while(resta >= entrada.getKey()){
                resultado = resultado + entrada.getValue();
                resta = resta - entrada.getKey();
            }
        }

        return resultado;
    }
}

File: PARK_C++/PARK_P3_EXAMEN.cpp
//
#include <iostream>
#include <map>
using namespace std;
string Romano(int numero);

int main()
{
    int numero;
    cout<<"*********************************"<<endl;
    cout<<"      DE ENTEROS A ROMANOS"<<endl;
    cout<<"*********************************"<<endl;
    cout<<"INGRESA UN NUMERO QUE NO EXCEDA 1001: "<<endl;
    cin >> numero;
    //sentencia que asegure que el numero sea menoa a 1001
    if(numero > 1001)
    {
        cout<<"ecxedes la limitante intenta otro numero: "<<endl;
        cin>> numero;
    }
    //se llama a la funcion para que se aplique la conbercion.
    string resultado = Romano(numero);
    cout<<"Tu numero es: "<<numero<<"a Romano es:"<<resultado<<endl;
    return 0;
}

string Romano(int numero)
{
    int resta = numero;
    string resultado = "";
    
    //se declara la funcion map
    map<int,string>rmo;
    rmo[1000] = "M";
    rmo[900] = "CM";
    rmo[500] = "D";
    rmo[400] = "CD";
    rmo[100] = "C";
    rmo[90] = "XC";
    rmo[50] = "L";
    rmo[40] = "XL";
    rmo[10] = "X";
    rmo[9] = "IX";
    rmo[5] = "V";
    rmo[4] = "IV";
    rmo[1] = "I";
    
    map<int, string>::reverse_iterator it = rmo.rbegin();
    
    while(it != rmo.rend())
    {
        if(resta >= it->first)
        {
            resultado = resultado + it->second;
            resta = resta - it->first;
        }else
        {
            it++;
        }
        
    }
    return resultado;
}